### PR TITLE
Add support for gp-suite in optee_test

### PR DESCRIPTION
**Pull request by jenswi-linaro**
By passing GP_PACKAGE=<path to GP-suite archive> optee_test is
configured to compile with the GlobalPlatform test suite. The test suite
is unpacked and patched in the build directory as part of the buildroot
process when building optee_test.

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
Needs to be tested and merged together with https://github.com/OP-TEE/optee_test/pull/449

**Review by jbech-linaro — COMMENTED** [review]
`Acked-by: Joakim Bech <joakim.bech@linaro.org>`

File: br-ext/package/optee_test_ext/Config.in [hunk 1]
```diff
@@ -24,4 +24,12 @@ config BR2_PACKAGE_OPTEE_TEST_EXT_CROSS_COMPILE
 	help
 	  some help
 
+config BR2_PACKAGE_OPTEE_TEST_EXT_GP_PACKAGE
+	string "GP package"
+	default ""
+	help
+	  Path to the GlobalPlatform package containing description files
+	  and TAs needed to build the complete GP test suite. The package
+	  is not free and has to be acquired directly from GlobalPlatform.
```
Comment (jforissier) [inline br-ext/package/optee_test_ext/Config.in:33]: Minor: you could be more precise here. The exact name of the package is "TEE Initial Configuration Test Suite with Excluded Tests List v2.0.0.2" and the URL is https://globalplatform.org/test-suites/tee-initial-configuration-test-suite-with-excluded-tests-list/. Also, the package is free for GP members and can be purchased by non-members. 

For clarity: I understand it is the path to the compressed archive (.7z file), no need to extract it first, right?

File: br-ext/package/optee_test_ext/optee_test_ext.mk [hunk 2]
```diff
@@ -6,10 +6,19 @@ OPTEE_TEST_EXT_INSTALL_STAGING = YES
 OPTEE_TEST_EXT_DEPENDENCIES = optee_client_ext openssl host-python3-pycryptodomex
 OPTEE_TEST_EXT_SDK = $(BR2_PACKAGE_OPTEE_TEST_EXT_SDK)
 OPTEE_TEST_EXT_CONF_OPTS = -DOPTEE_TEST_SDK=$(OPTEE_TEST_EXT_SDK)
-# os_test has dependencies, this enforces a valid build order
```
Comment (jforissier) [inline br-ext/package/optee_test_ext/optee_test_ext.mk:9]: Why remove the comment? I'm afraid that without it, someone might be tempted to replace the list with a simple wildcard `*`. In any case this does not belong to this PR.

**Review by jforissier — COMMENTED** [review]
Two small things below.
`Acked-by: Jerome Forissier <jerome@forissier.org>`

File: br-ext/package/optee_test_ext/Config.in (hunk 1 above)
Comment (jenswi-linaro) [reply br-ext/package/optee_test_ext/Config.in:33]: > For clarity: I understand it is the path to the compressed archive (.7z file), no need to extract it first, right?

Correct.

**Comment by jenswi-linaro** [comment]
Updated with comments addressed and tags applied.

**Comment by jforissier** [comment]
@jenswi-linaro I just recovered my access to the GlobalPlatform member area, so I could download the GP test suite and run it. It worked flawlessly. Good job! :+1: :wink:

**Comment by jenswi-linaro** [comment]
> @jenswi-linaro I just recovered my access to the GlobalPlatform member area, so I could download the GP test suite and run it. It worked flawlessly. Good job!

Nice. Thanks.

**Comment by jbech-linaro** [comment]
Which reminds me that I shall enable this for IBART. I'll probably do it during a weekend to avoid interference during regular working hours.

**Comment by jforissier** [comment]
@jbech-linaro yep, I noticed there are quite a lot of new tests so much better coverage I suppose.

**Comment by jbech-linaro** [comment]
I've enabled it for IBART for the Hikey device, it's working fine [[1]](https://github.com/OP-TEE/optee_test/pull/456) [[2]](https://optee.mooo.com:5000/logs/OP-TEE/optee_test/456/513692115/e11dfd910a7bb069f29d805c8136e61b3853df5f). Having said that I did 3 different builds locally. QEMUv7, QEMUv8 and Hikey. The two latter works, for former (QEMUv7, gives lots of compiler errors, like this:
```make
gp/include/xml_client_api.h:277:27: error: comparison of integer expressions of different signedness: ‘long int’ and ‘unsigned int’ [-Werror=sign-compare]                    
  277 |   } else if ((long)tmpMem != IGNORE) {\
      |                           ^~
gp_10000.c:28612:5: note: in expansion of macro ‘TEEC_initialize_memory’
28612 |     TEEC_initialize_memory(IGNORE, TEMP_MEM01, OFFSET0, SIZE_02, BYTE_04, BYTE_01, BYTE_02);
``` 
I'm currently using this locally on my machine:
```bash
$ ../toolchains/aarch32/bin/arm-linux-gnueabihf-gcc  --version
arm-linux-gnueabihf-gcc (GNU Toolchain for the A-profile Architecture 9.2-2019.12 (arm-9.10)) 9.2.1 20191025
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Related to this is also updated docs, see https://github.com/OP-TEE/optee_docs/pull/88.

Kudos to Jens for paving the way for this, for me it was a simple task to enable it in IBART.

**Comment by jenswi-linaro** [comment]
I'll take a look at those warning.
